### ISSUE TITLE
ESL Support

### ISSUE THREAD

**Issue by recklessnl** [matortheeternal/zedit#94]
I'm trying to patch a mod using a zEdit patcher. My load order has 290 mods. However, about 45 of these are patches in ESL format, so I'm still under the 254 limit when running the game.

However, I get errors when running zEdit, saying I'm exceeding the 254 plugin limit.

Will ESL support come to zEdit soon? Anything else I can do to fix this? ESLs are getting more popular by the day.

**Comment by matortheeternal** [comment]
~~[Using ESLs to bypass the 255 plugin limit is bad](https://old.reddit.com/r/skyrimmods/comments/9ieq24/does_everyone_realise_how_big_a_leap_forward/e6kipsw/).~~

~~zEdit may support ESLs at a future point in time, but patchers will continue to add all loaded plugins as masters as is required for fast execution times and well-defined failure conditions.   Load orders with >254 modules do not work with with full load order patching, so using ESLs to bypass the limit will continue to be an unfavorable workflow and I will not be going out of my way to support it.~~

~~Merging is a proven workflow that cleanly solves the problem of >254 modules in a clean and scalable fashion.  If you want to use full load order patching you will have to merge below 254 modules.~~

**Please see my latest comment below.**

**Comment by Qazerowl** [comment]
Your tools are the only ones that don't properly support ESL-flagged ESPs. Mod Oganizer, xEDIT and everything based on it, even Wyre Bash (which is now 10 years old) have all been updated to support ESL-flagged ESPs. Not to mention all future elder scrolls and fallout games will likely continue to support them.
Please reconsider your position. The dust has settled, and the rest of the modding community has accepted ESLs.

**Comment by Heartbreaker714** [comment]
+1 for this. The importance of ESL files is really growing. Some mods don't work right even when merged to copy everything. Particularly those involving NPCs on SSE. Please do reconsider

**Comment by matortheeternal** [comment]
Not sure why this issue was never updated, but a zEdit update for proper ESL support is on its way.  Included with this update will be a way for patchers to "select" required masters, allowing for UPF patchers to work with load orders containing >255 modules.

**Comment by jarrard** [comment]
> Not sure why this issue was never updated, but a zEdit update for proper ESL support is on its way. Included with this update will be a way for patchers to "select" required masters, allowing for UPF patchers to work with load orders containing >255 modules.

I look forward to this day.

**Comment by larinius** [comment]
I can say, without ESL support zEdit useless for me with 455 mods. Mods updated often and merging them is not an option.

**Comment by darkphoenixff4** [comment]
Would it be possible to give an ETA on this dropping?

**Comment by matortheeternal** [comment]
> Would it be possible to give an ETA on this dropping?

I would like to drop it with some other major things I've been working on, but I may be able to drop it sooner.  I'm reassessing my release plans now.

**Comment by Sunconure11** [comment]
Is there any place to view any progress on it so far, or would we have to wait?

**Comment by steve25469** [comment]
Yeah, I'm really looking forward to testing out the new 255+ support! :)  and have a perfect load order that I could really use it on right now.  lol   Glad to see that it's in the works in any case, great job man!

**Comment by abundzu** [comment]
Very excited for this feature to complement what is already an extraordinary set of tools. Thank you for all your work and patiently awaiting it being ready for public release!

**Comment by Heartbreaker714** [comment]
Just checking on if there was anymore progress... so excited for the esl limit thing being implemented. Especially since being stuck at home. lol
Just starting Skyrim again. Any more information on the status?

**Comment by Pleasantless** [comment]
still counting the days...

**Comment by DroidZed** [comment]
F
My mod list is continuously growing and the support of ESL is not here yet ....

**Comment by Heartbreaker714** [comment]
> F
> My mod list is continuously growing and the support of ESL is not here yet ....
Same here. Really hoping for a release soon.  I just found [THIS Mod script](https://www.nexusmods.com/skyrimspecialedition/mods/35012) recently put out as temporary workaround. I haven't tried it yet

**Comment by jarrard** [comment]
Obviously the huge amount of free time this virus has given everyone to work on projects hasn't helped zedit much. This leaves me to believe the developer is not really that interested in the project, so might need to rely on fork updates in the future. 

This isn't uncommon for github projects where the original dev(s) loose interest in a project and its picked up by someone else in the community. So if you know programming or can understand the code to this project it might be worth dabbling around in it, see what happens.

**Comment by Pleasantless** [comment]
we just need a final update so zEdit supports ESL files.  if he needs donations or whatever many are ready.

**Comment by Heartbreaker714** [comment]
> we just need a final update so zEdit supports ESL files. if he needs donations or whatever many are ready.

I know I'm one of them. Hell, it'd be worth it. Plus the guy literally answers every question people ask him on discord. That's some serious dedication

I'm going to try that link out soon to see if the work around works.

**Comment by DroidZed** [comment]
> > F
> > My mod list is continuously growing and the support of ESL is not here yet ....
> > Same here. Really hoping for a release soon.  I just found [THIS Mod script](https://www.nexusmods.com/skyrimspecialedition/mods/35012) recently put out as temporary workaround. I haven't tried it yet

Thanks for the link, Imma try it right now and see for myself, the steps look pretty straightforward and easy to follow.

**Comment by Heartbreaker714** [comment]
@DroidZed 
I keep getting distracted so still haven’t tried it yet. Did work for you?

**Comment by PsychoWookiee** [comment]
The workaround worked well for me. It should be adequate until z-Edit ESL support comes.

**Comment by DroidZed** [comment]
@Heartbreaker714 The patching was done successfully and I have to test it in game.

**Comment by DroidZed** [comment]
The only issue here is the number of UPF patches, if you have a lot then good luck in that....zEdit doesn't provide a search method to search the ESP individually in the ESP selection screen...sight

**Comment by Heartbreaker714** [comment]
> The only issue here is the number of UPF patches, if you have a lot then good luck in that....zEdit doesn't provide a search method to search the ESP individually in the ESP selection screen...sight

Well, at least it'll help for now. Hopefully, there will be ESL support in zEdit soon.

**Comment by matortheeternal** [comment]
> The only issue here is the number of UPF patches, if you have a lot then good luck in that....zEdit doesn't provide a search method to search the ESP individually in the ESP selection screen...sight

Ctrl+F

**Comment by DroidZed** [comment]
> > The only issue here is the number of UPF patches, if you have a lot then good luck in that....zEdit doesn't provide a search method to search the ESP individually in the ESP selection screen...sight
> 
> Ctrl+F

Cool thanks for that !

**Comment by DroidZed** [comment]
Anything new ?

**Comment by DroidZed** [comment]
Any news about the program ??

**Comment by Pleasantless** [comment]
i just accepted that we will never get it. if you need help with creating the patches i will as it is very simple.

**Comment by jarrard** [comment]
Most my patches are painstakingly made myself with xedit these days.  Half of all my plugins are flagged as ESL 👍 

Unfortunately FO4 has broken memory management so the only way I can play coherently is using SpecialK which has caching system. Can't wait for NVCache to be a thing with next NV cards! 

One way for zEdit to support ESL is just to remove the plugin limit all together and assume the user knows what their doing? hehe

Anyway I could probably do that and add in support for ESL myself, it is a git project afterall.  The problem I have is what does zEDIT have over xEDIT these days? xEDIT has been continuously updated!  Help me out here!

**Comment by jarrard** [comment]
To answer my own question:    https://z-edit.github.io/#/docs?t=Overview%2FxEdit_Comparison

However I rather help a project that uses something like python vs Javascript... But perhaps there is a good reason to NOT use python _(which I would want to use throughout the ENTIRE project)._

**Comment by recklessnl** [comment]
@matortheeternal I created this issue 18 months ago - can you give everybody a status update or ETA? 

This is by far the most requested feature. Would love to have it finally be added.

**Comment by matortheeternal** [comment]
I mostly finished the xEdit backend update which would include ESL support, but I encountered a weird bug with how record element nodes were built and got kind of sick of working with the xEdit codebase because of it (that section of code annoys the heck out of me).  So I started working on a project to totally replace it which has been what I've been primarily working on.  This new codebase will not have any plugin limitation when merging (doesn't matter if your plugins are ESL or not ESL or if you're merging for a game that supports or doesn't support ESLs).  This will be accomplished by not emulating global load order during the merging process.

https://github.com/matortheeternal/esper

There's still a bit of work to be done but it'll probably be finished in the next month or so, at which point I'll make an API wrapper equivalent to `xelib` so it can be a drop-in replacement for zEdit.

**Comment by KontraBute** [comment]
the struggles of coding... still waiting for your work.

**Comment by DroidZed** [comment]
thanks really, I'd be waiting for the full release !

**Comment by Sixdd6** [comment]
Just curious what the status is of this new system? I've got a boatload of plugins and a zEdit script that is just begging for the >254 support.

**Comment by genderneutralnoun** [comment]
It's been several months... is there any sort of status update we could get on the new system? I find Skyrim modding far too frustrating because of the various dynamic patches I need in my load order, so I quit playing Skyrim for the sake of my free time and health until this update comes out, expecting it to only be a month or two. So any kind of estimate would be appreciated, even if it ends up being innaccurate.

**Comment by Heartbreaker714** [comment]
Everything is pointing towards [Synthesis](https://github.com/Mutagen-Modding/Synthesis/wiki) is the way to go now. Looks like _zEdit_ won't ever be ideal for patchers & large load orders

**Comment by DroidZed** [comment]
Is there any news ?

**Comment by nickkeane** [comment]
Any updates on ESL support @matortheeternal ?

**Comment by l4kr** [comment]
rip
